proposals/native-integers.md

```diff
@@ -0,0 +1,342 @@
+# Native-sized integers
+
+## Summary
+[summary]: #summary
+
+Language support for a native-sized signed and unsigned integer types.
+
+The motivation is for interop scenarios and for low-level libraries.
+
+## Design
+[design]: #design
+
+The identifiers `nint` and `nuint` are new contextual keywords that represent native signed and unsigned integer types.
+The identifiers are only treated as keywords when name lookup does not find a viable result at that program location.
+```C#
+nint x = 3;
+var y = nameof(nuint);
+var z = nint.Zero;
+```
+
+The types `nint` and `nuint` are represented by the underlying types `System.IntPtr` and `System.UIntPtr` with compiler surfacing additional conversions and operations for those types as native ints.
+
+### Constants
+
+There is no direct syntax for native int literals. Explicit casts of other integral constant values can be used instead: `(nint)42`.
```

Comment:
>direct syntax for native int literals [](start = 12, length = 37)

Do you intend that constant expressions can be of these new types?  If so, you need to explicitly say so, as the spec for "constant expressions" excludes these types from the list of types for which constants exist.